docs/source/cite.rst

```diff
@@ -0,0 +1,22 @@
+Citations
```

Comment:
Good idea.  

This also starts the idea of maybe we should be referencing algorithms that we use or papers that were written with dask though.